state/etcd/etcd.go

```diff
@@ -141,9 +144,25 @@ func (e *Etcd) Get(ctx context.Context, req *state.GetRequest) (*state.GetRespon
 		return &state.GetResponse{}, nil
 	}
 
+	var metadata map[string]string
+	if resp.Kvs[0].Lease > 0 {
+		leaseResp, err := e.client.TimeToLive(ctx, clientv3.LeaseID(resp.Kvs[0].Lease))
```

Comment:
Does this make a network call?

Reply:
Yes, but there is no other way around this.

Reply:
I am concerned the perf impact in this case is not insigificant if we need to make an extra roundtrip to the database

Reply:
Indeed, I agree. It's also not even taking the now() clock time of the etcd process.